FreeRTOS-Plus/Demo/MQTT_Mutual_Auth_Qemu_mps2/Makefile

```diff
@@ -0,0 +1,222 @@
+KERNEL_DIR := $(FREERTOS_DIR)/Source
+
+FREERTOS_PLUS_DIR_REL := ../../../FreeRTOS-Plus
+FREERTOS_PLUS_DIR := $(abspath $(FREERTOS_PLUS_DIR_REL))
+
+SOURCE_FILES += init/startup.c  syscall.c main.c
+SOURCE_FILES += $(KERNEL_DIR)/portable/GCC/ARM_CM3/port.c
+SOURCE_FILES += $(KERNEL_DIR)/tasks.c
+SOURCE_FILES += $(KERNEL_DIR)/list.c
+SOURCE_FILES += $(KERNEL_DIR)/queue.c
+SOURCE_FILES += $(KERNEL_DIR)/timers.c
+SOURCE_FILES += $(KERNEL_DIR)/event_groups.c
+SOURCE_FILES += ${KERNEL_DIR}/portable/MemMang/heap_3.c
+
+INCLUDE_DIRS += -I$(FREERTOS_PLUS_DIR)/Demo/MQTT_Mutual_Auth_Qemu_mps2
+INCLUDE_DIRS += -I$(FREERTOS_PLUS_DIR)/Demo/MQTT_Mutual_Auth_Qemu_mps2/CMSIS
+INCLUDE_DIRS += -I$(KERNEL_DIR)/include
+INCLUDE_DIRS += -I$(KERNEL_DIR)/portable/GCC/ARM_CM3
+INCLUDE_DIRS += -I.
+
+FREERTOS_TCP = ${FREERTOS_PLUS_DIR}/Source/FreeRTOS-Plus-TCP
+MBED_TLS = ${FREERTOS_PLUS_DIR}/ThirdParty/mbedtls
+MBED_TLS_UTIL = ${FREERTOS_PLUS_DIR}/Source/Application-Protocols/network_transport
+COREMQTT = ${FREERTOS_PLUS_DIR}/Source/Application-Protocols/coreMQTT
+UTILITIES = ${FREERTOS_PLUS_DIR}/Source/Utilities
+
+# FreeRTOS networking header directories
+INCLUDE_DIRS += -I${FREERTOS_TCP}/include
+INCLUDE_DIRS += -I${FREERTOS_TCP}/portable/Compiler/GCC
+
+
+# mbedTLS header directories
+INCLUDE_DIRS += -I${MBED_TLS_UTIL}/using_mbedtls
+INCLUDE_DIRS += -I${MBED_TLS_UTIL}/sockets_wrapper/freertos_plus_tcp
+INCLUDE_DIRS += -I${MBED_TLS}/include
+INCLUDE_DIRS += -I${MBED_TLS}/library
+INCLUDE_DIRS += -I${UTILITIES}/mbedtls_freertos 
+
+
+# coreMQTT header directories 
+INCLUDE_DIRS += -I${COREMQTT}/source/include
+INCLUDE_DIRS += -I${COREMQTT}/source/interface
+INCLUDE_DIRS += -I${UTILITIES}/backoff_algorithm/source/include
+INCLUDE_DIRS += -I${UTILITIES}/logging
+
+
+
+# FreeRTOS networking sources
+SOURCE_FILES +=  main_networking.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_DNS.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_DHCP.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_ARP.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_TCP_WIN.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_Stream_Buffer.c
+SOURCE_FILES += ${FREERTOS_TCP}/portable/BufferManagement/BufferAllocation_2.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_IP.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_TCP_IP.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_UDP_IP.c
+SOURCE_FILES += ${FREERTOS_TCP}/FreeRTOS_Sockets.c
+SOURCE_FILES += ${FREERTOS_TCP}/portable/NetworkInterface/MPS2_AN385/NetworkInterface.c
+SOURCE_FILES += ${FREERTOS_TCP}/portable/NetworkInterface/MPS2_AN385/ether_lan9118/smsc9220_eth_drv.c
+SOURCE_FILES += DemoTasks/MutualAuthMQTTExample.c
+
+
+# mbedTLS sources
+SOURCE_FILES += mbedtls_freertos_port.c 
+SOURCE_FILES += ${UTILITIES}/mbedtls_freertos/mbedtls_bio_freertos_plus_tcp.c
+SOURCE_FILES += ${MBED_TLS_UTIL}/using_mbedtls/using_mbedtls.c
+SOURCE_FILES += ${MBED_TLS_UTIL}/sockets_wrapper/freertos_plus_tcp/sockets_wrapper.c
+SOURCE_FILES += ${MBED_TLS}/library/threading.c 
```

Comment:
I would prefer that you reference the mbedtls Makefile and build it separately as a static library rather than include the entire file list here.